virtualmachine/vsphere/vm.go

```diff
@@ -720,22 +722,22 @@ func (vm *VM) RemoveDisk(vmdkFiles []string) error {
 			switch device := d.(type) {
 			case *types.VirtualDisk:
 				fileName := d.GetVirtualDevice().Backing.(types.BaseVirtualDeviceFileBackingInfo).GetVirtualDeviceFileBackingInfo().FileName
-				if strings.HasSuffix(fileName, vmdkName) {
+				if strings.HasSuffix(fileName, disk.DiskName) {
```

Comment:
if `disk.DiskName` is empty, HasSuffix will return true and the first disk will be removed. Change this to `fileName == disk.DiskName`.